src/webhooks/JenkinsWebhook.ts

```diff
@@ -0,0 +1,124 @@
+
+import {Project} from "@atomist/automation-client/project/Project";
+import {BuildSystem} from "../commands/AddWebhookToCI";
+
+import {addWebhookToPipelineSection, getPipelineSection} from "../../src/webhooks/DeclarativeJenkinsfileGrammar";
+import {Webhook} from "../../src/webhooks/Webhook";
+import {addWebhookToNodes} from "./ScriptedJenkinsfileGrammar";
+
+export class JenkinsWebhook implements Webhook {
+
+    public readonly buildSystem: BuildSystem = "jenkins";
+
+    private jenkinsfilePath: string = "Jenkinsfile";
+
+    public addWebhook(project: Project, url: string): Promise<Project> {
+        return project.findFile(this.jenkinsfilePath).then(jenkinsfile => {
+            return jenkinsfile.getContent().then(content => {
+                if (isAtomistConfiguredForNotification(content)) {
+                    return project;
+                }
+                let newContent;
+                const syntax = this.detectSyntax(content);
+                if (syntax === "declarative-pipeline") {
+                    const pipelineSection = getPipelineSection(content);
+                    newContent = addWebhookToPipelineSection(content, pipelineSection, url);
+                } else if (syntax === "scripted-pipeline") {
+                    newContent = addWebhookToNodes(content, url);
+                } else {
+                    const msg = `Failed to detect the Jenkins file syntax in '${jenkinsfile.path}'`;
+                    return Promise.reject(msg);
+                }
+                if (newContent !== content) {
+                    return jenkinsfile.setContent(newContent).then(() => project);
+                }
+                return project;
+            });
+        }, err => {
```

Comment:
Does this catch all errors or should there be a catch at the end?

Reply:
I don't understand the question. "All errors" is ambiguous. I don't see any reason to have a catch, there is no try and no errors are specifically thrown. It will catch the error when a file is not found and return a message that is better than the default. But maybe I am not seeing something that you are.